src/vmprof_common.h

```diff
@@ -11,6 +11,8 @@ static int opened_profile(char *interp_name, int memory);
 static struct profbuf_s *volatile current_codes;
 #endif
 
+#define ENABLE_GZIP (defined(__unix__) || defined(__APPLE__))
```

Comment:
Why not simply use gzip all the time, and make it an implementation detail of the profile format?